packages/mui-material/src/AvatarGroup/AvatarGroup.js

```diff
@@ -127,6 +128,7 @@ const AvatarGroup = React.forwardRef(function AvatarGroup(inProps, ref) {
             marginLeft,
           }}
           variant={variant}
+          {...componentsProps.additionalAvatar}
```

Comment:
One thing I noticed during testing is that the provided componentsProps override the existing className. 

We should merge the class names instead. Same goes for the `style` prop:

```jsx
<AvatarGroupAvatar
  ownerState={ownerState}
  variant={variant}
  {...componentsProps.additionalAvatar}
  className={clsx(classes.avatar, componentsProps.additionalAvatar?.className)}
  style={{ marginLeft, ...componentsProps.additionalAvatar?.style }}
>
```

Reply:
that makes sense